sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -310,3 +312,28 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
     defineCodeGen(ctx, ev, c => s"($c).length()")
   }
 }
+
+/**
+ *  A function that return soundex name of a ID
+ */
+case class SoundEx(child: Expression) extends UnaryExpression with ExpectsInputTypes{
+  val so = new Soundex
```

Comment:
Seems the `Soundex` is not serializable, we have to make it as `@transient lazy val`